Mage/src/main/java/mage/abilities/keyword/TrainingAbility.java

```diff
@@ -61,32 +65,32 @@ public String getRule() {
     }
 }
 
-class TrainingAbilityEffect extends OneShotEffect {
+class TrainingWatcher extends Watcher {
 
-    TrainingAbilityEffect() {
-        super(Outcome.Neutral);
-    }
-
-    private TrainingAbilityEffect(final TrainingAbilityEffect effect) {
-        super(effect);
+    public TrainingWatcher() {
+        super(WatcherScope.GAME);
     }
 
     @Override
-    public TrainingAbilityEffect copy() {
-        return new TrainingAbilityEffect(this);
-    }
+    public void watch(GameEvent event, Game game) {
+        // 20240202 - 702.149c
+        // Some creatures with training have abilities that trigger when they train.
+        // "When this creature trains" means "When a resolving training ability puts a +1/+1 counter on this creature."
+        if (event.getType() == GameEvent.EventType.COUNTER_ADDED && event.getData().equals(CounterType.P1P1.getName())) {
+            StackObject stackObject = game.getStack().getStackObject(event.getSourceId());
+            if (!(stackObject instanceof StackAbility)) {
+                return;
+            }
 
-    @Override
-    public boolean apply(Game game, Ability source) {
-        Permanent permanent = source.getSourcePermanentIfItStillExists(game);
-        if (permanent == null) {
-            return false;
+            Ability ability = stackObject.getStackAbility();
+            if (ability instanceof TrainingAbility) {
+                game.fireEvent(GameEvent.getEvent(
```

Comment:
See original comment in another PR about simultaneous events instead fire:
https://github.com/magefree/mage/pull/11716#discussion_r1471141818